package polar_express;

// Next tag: 7
message Attributes {
  optional int64 id = 1;

  optional string owner_user = 2;
  optional string owner_group = 3;
  optional int32 uid = 4;
  optional int32 gid = 5;
  optional int32 mode = 6;
}

// Next tag: 1
message ExtraAttributes {
  // TODO: Add Symlink info, Windows ACLs, etc. here.
}

// Next tag: 13
message Snapshot {
  optional int64 id = 1;

  optional int64 file_id = 2;
  optional Attributes attributes = 3;

  optional int64 creation_time = 4;
  optional int64 modification_time = 5;
  optional int64 access_time = 6;
  optional ExtraAttributes extra_attributes = 7;
  optional bool is_regular = 8;
  optional bool is_deleted = 9;
  optional string hash = 10;
  optional int64 length = 11;

  optional int64 observation_time = 12;
}
